pkgs/applications/editors/pulsar/default.nix

```diff
@@ -0,0 +1,159 @@
+{ lib
+, stdenv
+, git
+, runtimeShell
+, fetchzip
+, wrapGAppsHook
+, glib
+, gtk3
+, atomEnv
+, xorg
+, libxkbcommon
+, hunspell
+, hunspellDicts
+, useHunspell ? true
+, languages ? [ "en_US" ]
+, makeDesktopItem
+, copyDesktopItems
+, makeWrapper
+}:
+
+let
+  pname = "Pulsar";
+  version = "1.103.0";
+
+  sourcesPath = {
+    x86_64-linux = {
+      tarname = "Linux.${pname}-${version}.tar.gz";
+      hash = "sha256-O+mekV2h3mxRPYpIrpoPHQyRDuXgl+En8n8u2yBG8TQ=";
+    };
+    aarch64-linux = {
+      tarname = "ARM.Linux.${pname}-${version}-arm64.tar.gz";
+      hash = "sha256-gajIHUJrjCogcJjJVSKz85/x9inuS9rEMYlLqcolUqg=";
+    };
+  }.${stdenv.hostPlatform.system} or (throw "Unsupported system: ${stdenv.hostPlatform.system}");
+
+  additionalLibs = lib.makeLibraryPath [
+    xorg.libxshmfence
+    libxkbcommon
+    xorg.libxkbfile
+  ];
+  newLibpath = "${atomEnv.libPath}:${additionalLibs}";
+
+  # Hunspell
+  hunspellDirs = builtins.map (lang: "${hunspellDicts.${lang}}/share/hunspell") languages;
+  hunspellTargetDirs = "$out/opt/Pulsar/resources/app.asar.unpacked/node_modules/spellchecker/vendor/hunspell_dictionaries";
+  hunspellCopyCommands = lib.concatMapStringsSep "\n" (lang: "cp -r ${lang}/* ${hunspellTargetDirs};") hunspellDirs;
+
+  desktopItem = makeDesktopItem {
+    name = "Pulsar";
+    desktopName = "Pulsar";
+    exec = "pulsar";
+    icon = "pulsar";
+    comment = "A Community-led Hyper-Hackable Text Editor";
+    genericName = "Text Editor";
+    categories = [ "Development" "TextEditor" "Utility" ];
+    mimeTypes = [ "text/plain" ];
+  };
+in
+stdenv.mkDerivation rec {
+  inherit pname version;
+
+  src = with sourcesPath; fetchzip {
+    url = "https://github.com/pulsar-edit/pulsar/releases/download/v${version}/${tarname}";
+    inherit hash;
+  };
+
+  patches = [
+    ./001-patch-wrapper.patch
+  ];
+
+  patchFlags = [ ];
```

Comment:
```suggestion
```
If it is empty, at some news to set it

Reply:
The empty patchFlags is there to override the default `[ "p1" ]`